### ISSUE TITLE
Prow `/approve` not cleared on push

### ISSUE THREAD

**Issue by tumido** [thoth-station/thoth-application#1215]
**Describe the bug**
If a contributor force-pushes or pushes into a PR which has an `/approve` label on it, this label is not cleared - the `approve` plugin of Prow doesn't support this. This is in contrast with `/lgtm` where this label get's cleared. See linked issues for details.

This issue is created to raise awareness of this + seek guidance on how we should use the `/approve` command... Our current setup merges a PR if there's an `approve` label, no matter if there's an `lgtm`.

**To Reproduce**
See: https://github.com/operate-first/operate-first-twitter/pull/10

**Expected behavior**
Approve label is either removed or the merge is blocked.

**Additional context**
Upstream bug (closed as WONTFIX): https://github.com/kubernetes/test-infra/issues/17698
https://github.com/operate-first/operate-first-twitter/pull/7#event-4657168525

**Comment by tumido** [comment]
@goern it merged the https://github.com/operate-first/operate-first-twitter/pull/10 PR :confused:

**Comment by goern** [comment]
Well, I would go with https://github.com/kubernetes/test-infra/issues/17698#issuecomment-633617368 Is there a reason we want to deviate from the kubernetes behavior/workflow?

/label triage/needs-information

**Comment by tumido** [comment]
@goern the thing is that we probably don't understand the kubernetes behavior/workflow in detail enough to follow it to point. For me, this behavior is an unpleasant surprise that exposes approved PRs to be exploited to get nonapproved changes into the repo.

This made me think if we have anything to do about this or if we're lacking any documentation in place how we should handle this. I have no idea if there's a code-change required here or rather a policy change or it's just about raising awareness of this and making sure that once a PR is approved it's monitored until merged to make sure there's no commits comming in (which still leaves the loophole exposed, but at least we know about it)...

**Comment by tumido** [comment]
Since `/lgtm` gets cleared on commit push, can we make it a requirement for tide to consider a PR for merge? Like configure Tide to merge only if there's both `lgtm` and `approve` labels? That would solve the issue, since `lgtm` is cleared on commit push.

**Comment by tumido** [comment]
cc @HumairAK @4n4nd @durandom for awareness

**Comment by tumido** [comment]
attempting to fix this via https://github.com/thoth-station/thoth-application/pull/1216

**Comment by tumido** [comment]
It seems the change #1216 worked as intended. At least for `operate-first` org this is fixed.

Second test PR behaved properly: https://github.com/operate-first/operate-first-twitter/pull/12

**Comment by goern** [comment]
From our point of view we should follow the Kubernetes workflows as close as possible, this will give us the highest leverage on using Prow. I will close this on for the moment, if some new requirements come out of the ADR creation, feel free to re-open.

/close
